numpy/_core/src/umath/loops_trigonometric.dispatch.cpp

```diff
@@ -151,23 +154,36 @@ simd_sincos_f32(const float *src, npy_intp ssrc, float *dst, npy_intp sdst,
                 iquadrant = hn::Add(iquadrant, ones);
             }
             // blend sin and cos based on the quadrant
-            opmask_t sine_mask = hn::RebindMask(f32, hn::Eq(hn::And(iquadrant, ones), hn::Zero(s32)));
+            opmask_t sine_mask = hn::RebindMask(
+                    f32, hn::Eq(hn::And(iquadrant, ones), hn::Zero(s32)));
             cos = hn::IfThenElse(sine_mask, sin, cos);
 
             // multiply by -1 for appropriate elements
-            opmask_t negate_mask = hn::RebindMask(f32, hn::Eq(hn::And(iquadrant, twos), twos));
+            opmask_t negate_mask = hn::RebindMask(
+                    f32, hn::Eq(hn::And(iquadrant, twos), twos));
             cos = hn::MaskedSubOr(cos, negate_mask, zerosf, cos);
             cos = hn::IfThenElse(nnan_mask, cos, hn::Set(f32, NPY_NANF));
 
             if (sdst == 1) {
                 hn::StoreN(cos, f32, dst, len);
-            } else {
+            }
+            else {
                 hn::ScatterIndexN(cos, f32, dst, dst_index, len);
             }
         }
         if (!hn::AllTrue(f32, simd_mask)) {
+            static_assert(hn::MaxLanes(f32) <= 16,
+                          "The following fallback is not applicable for "
+                          "SIMD widths larger than 512 bits, or for scalable "
+                          "SIMD in general.");
             npy_uint64 simd_maski;
-            hn::StoreMaskBits(f32, simd_mask, (uint8_t*)&simd_maski);
+            hn::StoreMaskBits(f32, simd_mask, (uint8_t *)&simd_maski);
```

Comment:
A better approach is needed for libc fallback to properly handle scalable extensions. One potential improvement could be introducing a new intrinsic, such as `bool TestBit(V, int pos)`, which would also be endian-friendly. For now, I applied a quick fix below to address a big-endian bug, as passing a `uint64_t` as a `uint8_t` array obviously lead to accessing garbage bytes instead.